src/general/MainMenu.cs

```diff
@@ -752,6 +790,33 @@ private void WarnAboutNoEnabledMods()
         }
     }
 
+    private float TrackMenuPerformance(float delta)
+    {
+        var currentFrameRate = 1 / delta;
+
+        // If this is the first frame, do not use the average of the frame delta and 0
+        if (averageFrameRate == 0)
+            return currentFrameRate;
+
+        // Not an exact average by any means, but good enough for this purpose
+        return (averageFrameRate + currentFrameRate) / 2;
+    }
+
+    private void WarnAboutLowPerformance()
+    {
+        if (averageFrameRate < 30 && secondsInMenu >= 15)
```

Comment:
> The low performance warning should only open if another menu popup is not open already, that way it's ensured to not get in the way.

Is this already checked for somewhere, or will I have to implement that from scratch?

> I think the measuring should be constant and be able to trigger later for example when the player has been in the menu for a while looking at new game settings.

Figured. Gonna sort that out.

Reply:
You can have a look around in the main menu file, it's possible that I made one of the popups trigger if all the other ones are hidden, but otherwise just create a new method called something like `IsAnyPopupOpen` in the main menu and just `return thanksDialog.Visible || someOtherDialog.Visible ...` (I can't remember the exact variable names off the top of my head).